### PR TITLE
[NET_HANDLER] Merge UDP functionality into TCP connection

### LINKED ISSUE #216: [NET_HANDLER] Merge UDP connection with Dawn into TCP connection
The amount of data that we're sending to Dawn on the UDP connection is not enough to justify the UDP connection with Dawn; we should try to merge the UDP connection with the TCP connection and see if there is any noticeable increase in lag. If there is no lag, then keeping that design is preferable, because it simplifies `net_handler` greatly. It also simplifies lots of logic on Dawn's end when managing the connection with Runtime.

### LINKED ISSUE #231: [TEST] Investigate tc_150_1 failing on Travis
The Travis build on PR #227 has failed a couple of times on `tc_150_1`. The test passes appears to pass consistently on an actual Raspberry Pi. The build passes when restarted. Here is a sample log:

```
*********************** Starting Test: "keyboard_input" ************************
DEBUG @ NET_HANDLER	(Thu Sep  9 21:28:28 2021) Received connection request from 127.0.0.1:37400
DEBUG @ NET_HANDLER	(Thu Sep  9 21:28:28 2021) Starting Shepherd connection
INFO @ DEV_HANDLER	(Thu Sep  9 21:28:28 2021) DEV_HANDLER initialized.
DEBUG @ DEV_HANDLER	(Thu Sep  9 21:28:28 2021) Polling now for devices.
INFO @ EXECUTOR	(Thu Sep  9 21:28:28 2021) Executor initialized
INFO @ DEV_HANDLER	(Thu Sep  9 21:28:29 2021) Connected SimpleTestDevice (0x0000000000000014) from year 62!
DEBUG @ DEV_HANDLER	(Thu Sep  9 21:28:29 2021) Monitoring SimpleTestDevice (0x0000000000000014)
keyboard_input: check 1 passed
DEBUG @ NET_HANDLER	(Thu Sep  9 21:28:31 2021) entering TELEOP mode
keyboard_input: check 2 failed
************************** Expected Pressed Buttons: ***************************
0000000000000000000000001000000000000000000000000000000000000000
************************************* Got: *************************************
0000000000000000000000000100000000000000000000000000000000000000
```

### PR DESCRIPTION
**Pull request by levincent06**
All data sent and received over UDP is now done over the TCP connection with Dawn and Shepherd. This simplifies the Runtime system with little to no effect on latency.

- `udp_tcp_converter` is removed.
- `net_handler` no longer creates a UDP socket.
- Ports for Dawn/Shepherd are no longer defined because we don't check it, nor do we need to. When creating a Dawn or Shepherd client, we let the OS choose the port when creating the connection socket.
- When a TCP connection is created for each client, we spawn an "events" thread and a "poll" thread. 
  - the events thread waits on a `select()` for messages from clients, challenge socket, and log FIFO and responds accordingly.
  - the poll thread sends messages on an interval.
- `get_next_dev_data()` and output suppression has been refactored .

Closes #216
Closes #231

### CODE REVIEW

**Review by benliao1 — COMMENTED** [review]
This is really good work to all of us, I only have one comment we should discuss before merging.

File: tests/client/net_handler_client.c [hunk 1]
```diff
@@ -40,20 +49,6 @@ static int connect_tcp(robot_desc_field_t client) {
...
-    cli_addr.sin_addr.s_addr = htonl(INADDR_ANY);                                        // use any address set on this machine to connect
-
-    // bind the client side too, so that net_handler can verify it's the proper client
-    if ((bind(sockfd, (struct sockaddr*) &cli_addr, sizeof(struct sockaddr_in))) != 0) {
-        printf("bind: failed to bind listening socket to client port: %s\n", strerror(errno));
-        close(sockfd);
-        stop_net_handler();
-        exit(1);
-    }
-
```
[the remark this answers is not in the document]
Comment (levincent06) [reply tests/client/net_handler_client.c:56]: This is in reference to binding the client's socket's port to a specific number (Lines 43 to 56)

File: net_handler/tcp_conn.c [hunk 2]
```diff
@@ -459,12 +708,20 @@ void stop_tcp_conn(robot_desc_field_t client) {
...
+    // Cancel and join the events thread
+    if (pthread_cancel((client == DAWN) ? dawn_events_tid : shepherd_events_tid) != 0) {
+        log_printf(ERROR, "stop_tcp_conn: Failed to cancel TCP client events thread for %d: %s", client, strerror(errno));
     }
-    if (pthread_join(tid, NULL) != 0) {
-        log_printf(ERROR, "stop_tcp_conn: Failed to join on TCP client thread for %d: %s", client, strerror(errno));
+    if (pthread_join((client == DAWN) ? dawn_events_tid : shepherd_events_tid, NULL) != 0) {
+        log_printf(ERROR, "stop_tcp_conn: Failed to join on TCP client events thread for %d: %s", client, strerror(errno));
     }
 }
```
[the remark this answers is not in the document]
Comment (benliao1) [reply net_handler/tcp_conn.c]: Implemented in next commit

**Review by benliao1 — APPROVED** [review]
Looks good! After refactor of `tcp_conn.c` into `connection.c` and `message.c`

**Pull request merged**